File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int maxSum = 0;
        for(int i : cardPoints){
            maxSum += i;
        }

        int windowLen = cardPoints.length - k;
        int window = 0;
        for(int i=0; i < windowLen; i++){
            window += cardPoints[i];
        }
        int result = maxSum - window;
        int start = 0;
        int end = windowLen;
        while(end < cardPoints.length){
            window = window + cardPoints[end++] - cardPoints[start++];
            result = Math.max(result,maxSum - window);
        }

        return result;

    }
}

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: list[int], k: int) -> int:
        n = len(cardPoints)
        total = sum(cardPoints)

        if n == k:
            return total

        window = n - k
        curr = sum(cardPoints[:window])
        min_sum = curr

        for r in range(window, n):
            curr += cardPoints[r] - cardPoints[r - window]
            min_sum = min(min_sum, curr)

        return total - min_sum